src/Service/TrustedDevice/DateTime/ExpirationHelper.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * Copyright 2024 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Tiqr\Service\TrustedDevice\DateTime;
+
+use DateTime as CoreDateTime;
+use Surfnet\StepupBundle\DateTime\DateTime;
+use Surfnet\Tiqr\Service\TrustedDevice\Exception\InvalidAuthenticationTimeException;
+use Surfnet\Tiqr\Service\TrustedDevice\ValueObject\CookieValueInterface;
+use TypeError;
+
+class ExpirationHelper implements ExpirationHelperInterface
+{
+    private CoreDateTime $now;
+
+    public function __construct(
```

Comment:
This helper shares information also present in the Configuration object. Maybe re-use that configuration object here. And hardcode the `60` grace period there?

Reply:
Good idea. The graceperiod is pointless in this feature, so I removed it.